node/p2p/addrs.go

```diff
@@ -47,7 +47,9 @@ func AddrsFactory(announce []string, noAnnounce []string) func() (_ p2pconfig.Ad
 
 		return func(maListen []ma.Multiaddr) (out []ma.Multiaddr) {
 			// combine maListen and maAnnounce addresses
-			maAnnounce = append(maAnnounce, maListen...)
+			out = make([]ma.Multiaddr, len(maAnnounce), len(maAnnounce)+len(maListen))
+			copy(out[:len(maAnnounce)], maAnnounce[:])
```

Comment:
want to clarify: Were it useless appending or `maAnnounce` should be used in `out`?

Reply:
@jenyasd209, Yeah, it's useless, should be `maListen = append(maListen, maAnnounce...)`. Thanks again for finding this out!

Reply:
Can addresses from `maAnnounce` exist in `maNoAnnounce`?

Reply:
@jenyasd209, Technically yes, in practice no one will do this

Reply:
discussions can be closed?

Reply:
I was thinking you are going to change appending to `maListen`, but to out is also fine.

Reply:
@jenyasd209, On last thing here I care about, is readability. Consider the following:
```go
out = make([]ma.Multiaddr, 0, len(maAnnounce)+len(maListen)) // this can be even removed, but proper preallocation is good
out = append(out, maAnnounce)
```

Reply:
Does exactly the same as your code, but less bulky.